app/services/sign_in/logout_redirect_generator.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+require 'sign_in/logingov/service'
+
+module SignIn
+  class LogoutRedirectGenerator
+    attr_reader :user, :client_config
+
+    def initialize(user:, client_config:)
+      @user = user
+      @client_config = client_config
+    end
+
+    def perform
+      return unless logout_redirect_uri
+
+      if authenticated_with_logingov?
+        logingov_service.render_logout(encode_logout_redirect)
+      else
+        parse_logout_redirect_uri
+      end
+    end
+
+    private
+
+    def parse_logout_redirect_uri
+      URI.parse(logout_redirect_uri).to_s
+    end
+
+    def authenticated_with_logingov?
+      authenticated_credential == SAML::User::LOGINGOV_CSID
```

Comment:
Should we change this constant so it's not dependent on `vets-api`?
```suggestion
      authenticated_credential == Constants::Auth::LOGINGOV
```

Reply:
👍

Reply:
I also went ahead and updated the use of this constant throughout SiS code